spring-pulsar/src/test/java/org/springframework/pulsar/listener/PulsarListenerTests.java

```diff
@@ -275,22 +285,42 @@ void avroSchema() throws Exception {
 					Collections.emptyMap());
 			PulsarTemplate<User> template = new PulsarTemplate<>(pulsarProducerFactory);
 			template.setSchema(AvroSchema.of(User.class));
-			template.send("avro-topic", new User("Avi", 2));
+
+			for (int i = 0; i < 3; i++) {
+				template.send("avro-topic", new User("Avi", i));
+			}
 			assertThat(avroLatch.await(10, TimeUnit.SECONDS)).isTrue();
+			assertThat(avroBatchLatch.await(10, TimeUnit.SECONDS)).isTrue();
 		}
 
 		@Test
 		void keyvalueSchema() throws Exception {
 			PulsarProducerFactory<KeyValue<String, Integer>> pulsarProducerFactory = new DefaultPulsarProducerFactory<>(
 					pulsarClient, Collections.emptyMap());
 			PulsarTemplate<KeyValue<String, Integer>> template = new PulsarTemplate<>(pulsarProducerFactory);
-
 			Schema<KeyValue<String, Integer>> kvSchema = Schema.KeyValue(Schema.STRING, Schema.INT32,
 					KeyValueEncodingType.INLINE);
-
 			template.setSchema(kvSchema);
-			template.send("keyvalue-topic", new KeyValue<>("Kevin", 3));
```

Comment:
@alpreu I forgot to mention this... I see what you did here w/ the clever naming of the message payload to alpha-relate to the feature being tested (avro/avi, json/jason, keyvalue/kevin, protobuf/paul). Seems minor thing but its much better than "foo" or "msg-N" as it relates to test and could help in associating during debugging later etc..